Student Exam Scores dataset

In [44]:
import pandas as pd
import numpy as np
df = pd.read_csv("/content/student_exam_scores.csv")
df.head()


,student_id,hours_studied,sleep_hours,attendance_percent,previous_scores,exam_score
0,S001,8.0,8.8,72.1,45,30.2
1,S002,1.3,8.6,60.7,55,25.0
2,S003,4.0,8.2,73.7,86,35.8
3,S004,3.5,4.8,95.1,66,34.0
4,S005,9.1,6.4,89.8,71,40.3


In [45]:
df.describe()

,hours_studied,sleep_hours,attendance_percent,previous_scores,exam_score
count,200.000000,200.000000,200.000000,200.000000,200.000000
mean,6.325500,6.622000,74.830000,66.800000,33.955000
std,3.227317,1.497138,14.249905,15.663869,6.789548
min,1.000000,4.000000,50.300000,40.000000,17.100000
25%,3.500000,5.300000,62.200000,54.000000,29.500000
50%,6.150000,6.700000,75.250000,67.500000,34.050000
75%,9.000000,8.025000,87.425000,80.000000,38.750000
max,12.000000,9.000000,100.000000,95.000000,51.300000


Converting the exam_score  item's into list and counting

In [46]:
exam_score = df['exam_score'].tolist()
sample_size = len(exam_score)
print(f"The Sample Size of the Dataset is {sample_size}")


The Sample Size of the Dataset is 200


Mean of the Score exam

In [47]:
mean = df['exam_score'].sum()/sample_size
print(f"The Mean of the Exam Score is {mean}")

The Mean of the Exam Score is 33.955


Calculating the Standard Deviation and Variance

In [48]:
sum = 0
for i in exam_score:
  sum = sum + ((i-mean)**2)
variance = sum/(sample_size-1)
print(f"The Variance of the Exam Score is {variance}")
standard_deviation = variance**0.5
print(f"The Standard Deviation of the Exam Score is {standard_deviation}")


The Variance of the Exam Score is 46.097964824120616
The Standard Deviation of the Exam Score is 6.789548204712934


Calculating the Standard Error of the Exam Scores

In [49]:
std_error = standard_deviation/(sample_size**0.5)
print(f"The Standard Error of the Exam Score is {std_error}")

The Standard Error of the Exam Score is 0.4800935576745465


## Conclusion of Standard Error::
The standard error is 0.480, which shows that the calculated mean exam score is a reliable estimate of the population mean

### Calculating the Margin of Error of Exam Score
For this analysis, we have chosen a 95% confidence level.
The corresponding Z-value from the standard normal distribution (Z-table) is 1.96.

In [50]:
z = 1.96
ME = z * std_error
print(f"The Margin of Error of Exam Score is {ME}")

The Margin of Error of Exam Score is 0.9409833730421111


### Conclusion of Margin of Error
The Margin of Error (ME) of the exam scores is 0.941. This means that the sample mean exam score could vary by ±0.941 from the true population mean with 95% confidence.

### Calculating the Confident Interval(C.I)  of Exam Score

In [51]:
### Lower Limit of Confident Interval
CI_lower = mean - ME
## Upper limit of confident Interval
CI_upper = mean + ME
print(f"95% Confidence Interval: [{CI_lower}, {CI_upper}]")

95% Confidence Interval: [33.01401662695789, 34.89598337304211]


## Conclusion::
The 95% confidence interval of exam scores is 33.01 to 34.90, indicating the sample mean is a reliable estimate of the true population mean.